### DISCUSSION TITLE
.Net: How does semantic kernel support load balancing by using multiple Azure OpenAI endpoints?

### DISCUSSION THREAD

**Discussion by vishnugms** [microsoft/semantic-kernel#5774] [Q&A] [answered]
How does semantic kernel support load balancing by using multiple Azure OpenAI endpoints? I would like to route my chat completion requests to different registered endpoints depending on the load.

Pl. assist.

**Comment by markwallace-microsoft** [comment, marked as answer]
This [article](https://learn.microsoft.com/en-us/samples/azure-samples/azure-openai-apim-load-balancing/azure-openai-service-load-balancing-with-azure-api-management/) explains how to use [Azure API Management](https://learn.microsoft.com/en-us/azure/api-management/api-management-key-concepts) to load balance requests to multiple instances of the [Azure OpenAI Service](https://learn.microsoft.com/en-us/azure/ai-services/openai/overview).

You will need to customise the OpenAICient that the Semantic Kernel uses to call the Azure OpenAI Service. This [sample](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/KernelSyntaxExamples/Example52_CustomOpenAIClient.cs) shows how to do this.
